framework/src/parser/Parser.C

```diff
@@ -929,6 +968,48 @@ void Parser::setVectorParameter(const std::string & full_name, const std::string
   }
 }
 
+
+template<typename T>
+void Parser::setDoubleIndexParameter(const std::string & full_name, const std::string & short_name, InputParameters::Parameter<std::vector<std::vector<T> > >* param, bool in_global, GlobalParamsAction * global_block)
+{
+  GetPot *gp;
+
+  // See if this variable was passed on the command line
+  // if it was then we will retrieve the value from the command line instead of the file
+  if (_app.commandLine() && _app.commandLine()->haveVariable(full_name.c_str()))
+    gp = _app.commandLine()->getPot();
+  else
+    gp = &_getpot_file;
+
+  // Get the full string assigned to the variable full_name
+  std::string buffer = gp->get_value_no_default(full_name, "");
+
+  // split vector at delim ;
+  // NOTE: the substrings are _not_ of type T yet
+  std::vector<std::string> first_tokenized_vector;
+  MooseUtils::tokenize(buffer, first_tokenized_vector, 1, ";");
+  param->set().resize(first_tokenized_vector.size());
+
+  for (unsigned j = 0; j < first_tokenized_vector.size(); ++j)
+    if (!MooseUtils::tokenizeAndConvert<T>(first_tokenized_vector[j],param->set()[j]))
+      mooseError("Failed reading ... do some appropriate error handling.");
+
+  if (in_global)
+  {
```

Comment:
I know this is a WIP but this will definitely need to be filled out before you are done.